File: in_Java/0717 1-bit and 2-bit Characters.java
class Solution {
    public boolean isOneBitCharacter(int[] bits) {
        int i;
        for(i = 0; i < bits.length-1; ){
            if(bits[i] == 1) i += 2;
            else i++;
        }
        return i == bits.length-1;
    }
}

File: in_Python/0717 1-bit and 2-bit Characters.py
class Solution(object):
    def isOneBitCharacter(self, bits):
        """
        :type bits: List[int]
        :rtype: bool
        """
        i = 0
        loc = [0]
        while i < len(bits):
            if bits[i] == 1:
                i += 2
            else:
                i += 1
            loc.append(i)
        return len(bits)-1 in loc